sklearn/ensemble/_hist_gradient_boosting/loss.py

```diff
@@ -111,34 +144,40 @@ class LeastSquares(BaseLoss):
 
         loss(x_i) = 0.5 * (y_true_i - raw_pred_i)**2
 
-    This actually computes the half least squares loss to optimize simplify
+    This actually computes the half least squares loss to simplify
     the computation of the gradients and get a unit hessian (and be consistent
     with what is done in LightGBM).
     """
 
     hessians_are_constant = True
 
-    def __call__(self, y_true, raw_predictions, average=True):
+    def __call__(self, y_true, raw_predictions):
         # shape (1, n_samples) --> (n_samples,). reshape(-1) is more likely to
         # return a view.
         raw_predictions = raw_predictions.reshape(-1)
         loss = 0.5 * np.power(y_true - raw_predictions, 2)
-        return loss.mean() if average else loss
+        return loss
 
-    def get_baseline_prediction(self, y_train, prediction_dim):
-        return np.mean(y_train)
+    def get_baseline_prediction(self, y_train, sample_weight, prediction_dim):
+        if sample_weight is None:
+            return np.mean(y_train)
+        else:
+            return np.average(y_train, weights=sample_weight)
 
     @staticmethod
     def inverse_link_function(raw_predictions):
         return raw_predictions
 
     def update_gradients_and_hessians(self, gradients, hessians, y_true,
-                                      raw_predictions):
+                                      raw_predictions, sample_weight):
         # shape (1, n_samples) --> (n_samples,). reshape(-1) is more likely to
         # return a view.
         raw_predictions = raw_predictions.reshape(-1)
         gradients = gradients.reshape(-1)
         _update_gradients_least_squares(gradients, y_true, raw_predictions)
+        if sample_weight is not None:
+            np.multiply(gradients, sample_weight, out=gradients)
```

Comment:
Is it any different from `gradients *= sample_weight`? No strong preference anyway.

However, while this is OK as a quick first version, I think we'll want to handle the sample_weight in the Cython code parts, to take advantage of the parallelism. This is making the code sequential again, so it's going to be significantly slower.

Reply:
ok, moved them to the cython code.